### DISCUSSION TITLE
Minor Planet Orbits (Trails)

### DISCUSSION THREAD

**Discussion by Skipper197600** [Stellarium/stellarium#1868] [General]
Minor Planet Orbits 

This function does not work  (v0.21.1)

This is a really useful function. Is there a known issue?

**Comment by alex-w** [comment]
Orbits is orbits, trails is trails and this is different features.

**Reply by DavidSChandler** [reply]
I am interested in trails (the orbit as seen from a fixed point projected onto the sky as opposed to a daily updated ephemeris path) for comets.  For instance Comet Shoemaker-Levy 9 was a chain of fragments aligned along the orbital path.  This kind of feature would be easy to generate by holding the earth position fixed and allowing the comet to cycle through its orbit.  Would the minor planet trails feature work for this situation?

**Reply by alex-w** [reply]
Please check AstroCalc/Ephemeris tool

**Reply by DavidSChandler** [reply]
There are two particular applications I have in mind for this feature.  1. Recovery of comets that don't return on schedule.  2. Observation of the debris from a comet that is strewn along the orbital path.  I had this feature built into my software (Deep Space) back in the DOS era, but I would like to see this feature in a currently accessible, widely used program.  If this feature is already implemented, please point me to how to access it.

**Reply by alex-w** [reply]
I don't know what I can answer here. I never seen Deep Space software and I have no idea what exactly it was display. AstroCalc/Ephemeris tool in Stellarium may display ephemeris of Solar system bodies in equatorial and horizontal coordinates. Important note: ephemeris of SSO computes from elements of orbits (math. way) and Stellarium cannot use any physics models to prognose behavior of comets.

**Reply by DavidSChandler** [reply]
To get a geocentric position of a comet, you first compute heliocentric positions of the earth and comet, then construct a vector difference, which points from the earth to the comet.  This vector, when projected onto the sky, gives the geocentric position.  To get an ephemeris (or a daily map of where to find the comet), update the earth and comet each day and repeat the calculation.  However, to project the orbit onto the sky, hold the earth position fixed and update only the comet through its orbit, or a certain fraction of the orbit if it is parabolic or hyperbolic.  This is what I call a comet "trail", but I don't know if that is how you are using the term.  This is the path you would search along to find what material may be seen in the orbit of the comet.

**Reply by gzotti** [reply]
To visualize debris along the path you could add multiple copies of the comet with slightly different perihel dates. However this is a brutal hack...
What you need to display (in Stellarium's terms) is the orbit (rendering of the 3d path in space), not the path (projected positions in earth's sky).

**Comment by Skipper197600** [comment]
Ok...

Minor Planet Trails

This function does not work (v0.21.1)

This is a really useful function. Is there a known issue?

**Comment by alex-w** [comment]
Please check settings (F4 -> SSO -> Trails)

![stellarium-000](https://user-images.githubusercontent.com/88731/132130430-faca98c5-8abb-493f-8563-96b79d5adc76.jpg)

**Reply by Skipper197600** [reply]
Yes, that tick box is selected. :/

**Reply by alex-w** [reply]
Please write about your settings for this option and what exactly you doing to get a "nothing"? What about your operating system and mode of running the Stellarium? When you just said "This function does not work" then this doesn't help us to diagnosting of your problem and solve it (the feature is work - you may see it directly from my screenshot).

**Comment by DavidSChandler** [comment]
> What you need to display (in Stellarium's terms) is the orbit (rendering
> of the 3d path in space), not the path (projected positions in earth's sky).
>
No, what I need is the orbit projected onto the sky.  I want to be able to
take my telescope and point it to positions along the orbital path of the
comet.  I did this in my old DOS era software and the feature was very easy
to implement.  I did a loop incrementing the daily heliocentric location of
the comet for the portion of the orbit where it was less than about 5 AU
from the sun, while keeping the earth position fixed.  For each
heliocentric position of the comet and for the single heliocentric position
of the earth I did a vector difference, which creates a 3-D vector pointing
from the earth to a point on the comet's orbit.  I then plotted those
positions on my star map as a path where one could look for comet debris,
recover comets that did not return on schedule, etc.  (I called this comet
recovery mode, but I primarily used it for observing comet debris...which,
by the way, is observable in dark skies and sufficiently large telescopes.
I want to write about those observations, but not until that feature is in
some software I can point people to, so they can see them for themselves.)

**Reply by gzotti** [reply]
The rendering of the comet orbit is just that. For each frame you have earth's position plus a series of theoretical comet positions along its orbit, a.k.a. orbit line.

**Comment by DavidSChandler** [comment]
Hello Georg,
I have returned to this problem once again (working this time with Don
Machholz, as in Comet 141P/Machholz) to figure out how to get it right.  I
have downloaded Stellarium on both my Linux and Windows machines.  I have
so far been able to create the 3-D orbit (in Stellarium's terms) for major
planets, like Mars.  In attempting to show the orbit for 141P/Machholz I
have
--plotted an ephemeris for it on the sky for a range of dates,
--searched for it and have its current position marked
--chose the setting that says to show an orbit only for the selected
object, not for the major planets,
When I do all this I get orbits for multiple major planets and not the
selected comet.
See if you can successfully show the orbit.  Am I doing something wrong?
For the project I am working on it is important that this feature work
correctly.  --David Chandler

**Reply by gzotti** [reply]
There is an issue #1842 still open related to that. Sorry, something to work on in December weekends, earliest.
